### PR TITLE
feat: integrate executed block into OP built payload

### PR DESCRIPTION
**Pull request by mattsse**
towards https://github.com/paradigmxyz/reth/issues/10159

integrates the `ExecutedBlock` data into OP's built payload

### CODE REVIEW

File: crates/optimism/payload/src/builder.rs [hunk 1]
```diff
@@ -403,6 +408,7 @@ where
         }));
 
         // append transaction to the list of executed transactions
```
Comment (fgimenez) [inline crates/optimism/payload/src/builder.rs:410]:
```suggestion
        // append sender and transaction to the respective lists
```

File: crates/optimism/payload/src/builder.rs [hunk 2]
```diff
@@ -491,6 +497,7 @@ where
             total_fees += U256::from(miner_fee) * U256::from(gas_used);
 
             // append transaction to the list of executed transactions
```
Comment (fgimenez) [inline crates/optimism/payload/src/builder.rs:499]:
```suggestion
            // append sender and transaction to the respective lists
```

**Review by fgimenez — APPROVED** [review]
looks good, small doc nit

**Review by rkrasiuk — APPROVED** [review]
seems fine to me

**Pull request merged**